scheduler/pkg/kafka/pipeline/multi_topic_consumer.go

```diff
@@ -156,12 +156,17 @@ func (c *MultiTopicsKafkaConsumer) pollAndMatch() error {
 				Debugf("received message")
 
 			if val, ok := c.requests.Get(string(e.Key)); ok {
-				// Add tracing span
 				ctx := context.Background()
 				carrierIn := splunkkafka.NewMessageCarrier(e)
 				ctx = otel.GetTextMapPropagator().Extract(ctx, carrierIn)
+
+				// Add tracing span
 				_, span := c.tracer.Start(ctx, "Consume")
-				span.SetAttributes(attribute.String(util.RequestIdHeader, string(e.Key)))
+				requestId := getRequestIdFromKafkaHeaders(e.Headers)
```

Comment:
maybe add a note that the key is composite and cannot be used as the request id?